packages/vitest/src/node/reporters/json.ts

```diff
@@ -152,6 +160,8 @@ export class JsonReporter implements Reporter {
             t.result?.errors?.map(e => e.stack || e.message) || [],
           location: t.location,
           meta: t.meta,
+          retryCount: t.result?.retryCount ?? 0,
+          flaky: !!t.result?.retryCount && t.result?.state === 'pass' && t.result?.retryCount > 0,
```

Comment:
We should probably re-use the logic from `diagnostics()` here:

```diff
      const assertionResults = tests.map((t) => {
+       const entity = this.ctx.state.getReportedEntity(t)
+       const diagnostic = entity?.type === 'test' ? entity.diagnostic() : null
+       const flaky = diagnostic?.flaky
+       const retryCount = diagnostic?.retryCount
```

Reply:
Thanks for it, fixed